### ISSUE TITLE
Restore with wallet file

### ISSUE THREAD

**Issue by HardenedSteel** [haveno-dex/haveno#1048]
Is it possible to use/restore a wallet from a file instead creating new wallet? Simply replacing wallet file or replacing all files give error.

**Comment by woodser** [comment]
If your app is started from a fresh instance and you replace the wallet file, it should be fine. But if the app is used (funds deposited, trades made, etc), it can produce errors, since the app will depend on previous wallet state.

**Comment by HardenedSteel** [comment]
![image](https://github.com/haveno-dex/haveno/assets/19209715/e7f79938-8401-47d6-aa77-0dcad65b0460)

I get `Failed to open wallet error`, steps to reproduce;
1. Clear all data in `~/.local/share/Haveno/`
2. move wallet file to `~/.local/share/Haveno/xmr_mainnet/wallet`
3. rename wallet file to `haveno_XMR`
4. relaunch the app user agreement pops up, Tor connects and node creates
5. then the error

**Comment by woodser** [comment]
First close any running instances of monero-wallet-rpc, then start the app. Does the error still happen?

**Comment by HardenedSteel** [comment]
yes i already followed these steps while Haveno is closed.

**Comment by woodser** [comment]
Please attach a log file from the application startup.

**Comment by HardenedSteel** [comment]
oh sorry, I should've check the logs earlier, changing the wallet password to "password" solved the issue. This may need to change.

**Comment by woodser** [comment]
"password" is the default password for an unencrypted wallet.

Can this issue be closed then?

**Comment by HardenedSteel** [comment]
Shouldn't the error message state something like wrong password then ask the wallet's password?

**Comment by woodser** [comment]
Did you manually change the wallet's password? The application will expect the default "password" unless you changed the password within the app, which encrypts everything.

**Comment by HardenedSteel** [comment]
Yes, the password were different than default `password`. I had to change the password to import a wallet.

**Comment by woodser** [comment]
This would be expected behavior then, since the application is not expecting to have wallets with a different password than the rest of the application.

The error message "Failed to open wallet" comes from monero-wallet-rpc when an incorrect password is used. Maybe it would be best to open an issue on monero-project to provide a better error message so the problem is more clear.
